node/src/api/server.rs

```diff
@@ -3,18 +3,23 @@ use crate::{
     config::NodeConfig,
     consensus::ConsensusProcessor,
     db::utils::db::build_pool,
+    wallet::WalletStore,
 };
 use actix_cors::Cors;
-use actix_web::{http, middleware::Logger, web, App, HttpResponse, HttpServer};
+use actix_web::{http, middleware::Logger, web, App, HttpResponse, HttpServer, Scope};
 use serde_json::json;
-use std::{net::ToSocketAddrs, sync::mpsc};
+use std::{net::ToSocketAddrs, sync::Arc};
+use tokio::sync::Mutex;
 
 // Must be valid JSON
 const LOGGER_FORMAT: &'static str = r#"{"level": "INFO", "target":"api::request", "remote_ip":"%a", "user_agent": "%{User-Agent}i", "request": "%r", "uri": "%U", "status_code": %s, "response_time": %D, "api_version":"%{x-app-version}o", "client_version": "%{X-API-Client-Version}i" }"#;
 
-pub async fn actix_main<F>(config: NodeConfig, configure: F) -> anyhow::Result<()>
-where F: FnOnce(&mut web::ServiceConfig) + Send + Clone + 'static {
-    let pool = web::Data::new(build_pool(&config.postgres)?);
+pub async fn actix_main<F>(config: NodeConfig, scopes: F) -> anyhow::Result<()>
+where F: (FnOnce() -> Vec<Scope>) + Clone + Send + 'static {
+    let pool = Arc::new(build_pool(&config.postgres)?);
+    let wallets = WalletStore::init(config.wallets_keys_path.clone())?;
+    let wallets = Arc::new(Mutex::new(wallets));
+    let config_arc = Arc::new(config.clone());
```

Comment:
Server is still not working in this PR as it expects TemplateContext to be configured and available as web::Data for routes.